appscale/tools/node_layout.py

```diff
@@ -531,6 +536,16 @@ def is_valid_advanced_format(self):
     # and return a list of nodes or raise BadConfigurationException.
     nodes = self.distribute_unassigned_roles(node_hash.values(), role_count)
 
+    for node in nodes:
```

Comment:
This seems similar to the check in the simple format: should it be its own function/method/library call?

Reply:
I could make it a part of distribute_unassigned_roles